judo-tatami-jsl-jsl2psm/src/main/java/hu/blackbelt/judo/tatami/jsl/jsl2psm/JslExpressionToJqlExpression.java

```diff
@@ -71,55 +76,59 @@ public static String getJqlForDerived(EntityDerivedDeclaration declaration, Stri
-                                e -> e.getName(), 
-                                e -> e.getDefault() != null ? "(input." + e.getName() + "!isDefined() ? " 
-                                        + "input." + e.getName() 
-                                        + " : " + transformer.getJql(e.getDefault()) + ")" : "input." + e.getName(), 
-                                (key1, key2)-> key2)));
-
-        transformer.queryStackParameterValues.add(parameterValues);
+ 
         transformer.entityNamePrefix = entityNamePrefix;
         transformer.entityNamePostfix = entityNamePostfix;
-        
-        return transformer.getJql(declaration.getExpression());
+
+    	HashMap<String, String> passedArgs = new HashMap<String, String>();
+    	
+    	for (QueryParameterDeclaration queryParameter : declaration.getParameters()) {
+    		String value = "input." + queryParameter.getName();
+    		
+    		if (queryParameter.getDefault() != null) {
+    			value = String.format("%s!isDefined() ? %s : %s", value, value, transformer.getJql(queryParameter.getDefault()));
+    		}
+    		
+    		passedArgs.put(queryParameter.getName(), value);
+    	}
+    	
+        return transformer.getJql(declaration.getExpression(), passedArgs);
     }
 
 
     public static String getJqlForStaticQuery(QueryDeclaration declaration, String entityNamePrefix, String entityNamePostfix) {
         JslExpressionToJqlExpression transformer = new JslExpressionToJqlExpression();
 
-        Map<String, String> parameterValues = new HashMap();
-        parameterValues.putAll(declaration.getParameters()
-                .stream()
-                .collect(
-                        Collectors.toMap(
-                                e -> e.getName(), 
-                                e -> e.getDefault() != null ? "(input." + e.getName() + "!isDefined() ? " 
-                                        + "input." + e.getName() 
-                                        + " : " + transformer.getJql(e.getDefault()) + ")" : "input." + e.getName(), 
-                                (key1, key2)-> key2)));
-
-        transformer.queryStackParameterValues.add(parameterValues);
         transformer.entityNamePrefix = entityNamePrefix;
         transformer.entityNamePostfix = entityNamePostfix;
-        transformer.selfEnabled = false;
 
-        return transformer.getJql(declaration.getExpression());
+    	HashMap<String, String> passedArgs = new HashMap<String, String>();
+    	
+    	for (QueryParameterDeclaration queryParameter : declaration.getParameters()) {
+    		String value = "input." + queryParameter.getName();
+    		
+    		if (queryParameter.getDefault() != null) {
+    			value = String.format("%s!isDefined() ? %s : %s", value, value, transformer.getJql(queryParameter.getDefault()));
+    		}
+    		
+    		passedArgs.put(queryParameter.getName(), value);
+    	}
+    	
+        return transformer.getJql(declaration.getExpression(), passedArgs);
```

Comment:
Why the parentheses is removed from logic? And what was the problem with the lambda collector? There can be function call on the value.

Reply:
This return string is never used in oter query. However, parenthesis are needed in "then" and "else" statements. Fixed.